storm/src/main/storm/mesos/util/PrettyProtobuf.java

```diff
@@ -176,51 +224,49 @@ private static String rangeToString(Range range) {
   }
 
   /**
-   * Pretty-print mesos protobuf Ranges.
+   * Pretty-print List of mesos protobuf Ranges.
    */
-  private static String rangesToString(Ranges ranges) {
-    List<String> rangesAsStrings = Lists.transform(ranges.getRangeList(), rangeToStringTransform);
+  private static String rangeListToString(List<Range> ranges) {
+    List<String> rangesAsStrings = Lists.transform(ranges, rangeToStringTransform);
     return "[" + StringUtils.join(rangesAsStrings, ",") + "]";
   }
 
   /**
-   * Pretty-print mesos protobuf Set.
-   */
-  private static String setToString(Set set) {
-    return "[" + StringUtils.join(set.getItemList(), ",") + "]";
-  }
-
-  /**
-   * Return Resource names mapped to values.
+   * Construct a Map of Resource names to String values.
+   * Ensure the order is always the same (cpu, mem, then ports), so that
+   * we have consistently ordered log output.
    */
   private static Map<String, String> resourcesToOrderedMap(List<Resource> resources) {
-    Map<String, String> map = new TreeMap<>();
+    String cpus = null;
+    String mem = null;
+    String ports = null;
+
     for (Resource r : resources) {
-      String name;
-      String value = "";
-      if (r.hasRole()) {
-        name = r.getName() + "(" + r.getRole() + ")";
-      } else {
-        name = r.getName();
-      }
-      switch (r.getType()) {
-        case SCALAR:
-          value = String.valueOf(r.getScalar().getValue());
+      switch (r.getName()) {
+        case "cpus":
+          cpus = String.valueOf(r.getScalar().getValue());
           break;
-        case RANGES:
-          value = rangesToString(r.getRanges());
+        case "mem":
```

Comment:
Definitely wrong.  The implementation in storm-mesos is better than what you are copying into here.  The existing implementation is more generic.  Please revert.